### PR TITLE
remove evmjit

### LINKED ISSUE #6205: evmjit update
- it should use https://github.com/ethereum/evmjit
- it should not be excluded from CI https://github.com/paritytech/parity/pull/6187#discussion_r130561517
- cargo should automatically detect if evmjit is available and if tests should be run

### LINKED ISSUE #8192: Can't build ubuntu-jit image
_Before filing a new issue, please **provide the following information**._

> I'm running:
>
> - **Which Parity version?**: 0.0.0
current master branch

> - **Which operating system?**: Windows / MacOS / Linux
Linux

> - **How installed?**: via installer / homebrew / binaries / from source
building from source

`docker build -f docker/ubuntu-jit/Dockerfile --tag ethcore/parity:ptx .`

```
Compiling evmjit v1.11.0 (file:///build/parity/evmjit)
     Running `rustc --crate-name evmjit evmjit/src/lib.rs --crate-type dylib --emit=dep-info,link -C prefer-dynamic -C opt-level=3 -C panic=abort -C metadata=b5f143990e7b6fba --out-dir /build/parity/target/release/deps -L dependency=/build/parity/target/release/deps --extern tiny_keccak=/build/parity/target/release/deps/libtiny_keccak-4edaa4c1facdfc5b.rlib`
error: the linked panic runtime `panic_unwind` is not compiled with this crate's panic strategy `abort`

error: aborting due to previous error

error: Could not compile `evmjit`.

Caused by:
  process didn't exit successfully: `rustc --crate-name evmjit evmjit/src/lib.rs --crate-type dylib --emit=dep-info,link -C prefer-dynamic -C opt-level=3 -C panic=abort -C metadata=b5f143990e7b6fba --out-dir /build/parity/target/release/deps -L dependency=/build/parity/target/release/deps --extern tiny_keccak=/build/parity/target/release/deps/libtiny_keccak-4edaa4c1facdfc5b.rlib` (exit code: 101)
warning: build failed, waiting for other jobs to finish...
error: build failed
The command '/bin/sh -c cd parity &&         cargo build --release --features ethcore/jit --verbose &&         ls /build/parity/target/release/parity &&               strip /build/parity/target/release/parity' returned a non-zero code: 101
```

### PR DESCRIPTION
**Pull request by debris**
removed not working evmjit, closes #8192 closes #6205

### CODE REVIEW

**Comment by debris** [comment]
cc @tomusdrw

File: ethcore/evm/src/factory.rs [hunk 1]
```diff
@@ -101,24 +74,7 @@ fn test_create_vm() {
...
-			$name_test(Factory::new(VMType::Interpreter, 1024 * 32));
-		}
-	};
-	($name_test: ident: $name_jit: ident, $name_int: ident) => {
-		#[test]
-		#[cfg(feature = "jit")]
-		fn $name_jit() {
-			$name_test(Factory::new(VMType::Jit, 1024 * 32));
-		}
+	($name_test: ident: $name_int: ident) => {
```
Comment (tomusdrw) [inline ethcore/evm/src/factory.rs]: This could be further removed (inlined), since now we only create one test.

Comment (debris) [reply ethcore/evm/src/factory.rs]: should I just remove this macro completely? Cause this is the only way to 'inline' this test further. Please note, that this macro creates function with name `$name_int` which calls `$name_test`.

Comment (tomusdrw) [reply ethcore/evm/src/factory.rs]: Yeah, that's what I thought.

**Pull request merged**